hotness/buildsys.py

```diff
@@ -163,60 +166,46 @@ def handle(self, package, upstream, version, rhbz):
-        """
-        Rebase helper part which does a rebase a inform package
-        maintainer whether package was rebased properly.
-        Output information are in dictionary rh_stuff.
+    def _run_rh(self, package, args, tmp_dir):
+        if package:
+            url = self.git_url.format(package=package)
+            self.log.info("Cloning %r to %r" % (url, tmp_dir))
+            sh.git.clone(url, tmp_dir)
 
-        """
-        self.log.info("Rebase-helper is going to rebase package")
-        rh_stuff = {}
-        result_rh = -1
-        rh_app = None
-        url = self.git_url.format(package=package)
-        self.log.info("Cloning %r to %r" % (url, tmp))
-        sh.git.clone(url, tmp)
-        os.chdir(tmp)
+        cwd = os.getcwd()
+        os.chdir(tmp_dir)
         try:
-            argument = ['--non-interactive', '--builds-nowait', '--buildtool', 'koji', upstream]
-            LoggerHelper.add_stream_handler(logger, logging.INFO)
-            cli = CLI(argument)
+            cli = CLI(args)
             rh_app = Application(cli, *Application.setup(cli))
             rh_app.set_upstream_monitoring()
-            self.log.info("Rebasehelper package %s %s" % (package, upstream))
+            self.log.info("Running rebase-helper with arguments '%s'" % ' '.join(args))
             result_rh = rh_app.run()
-            if result_rh is None:
+            if result_rh is not None:
+                result_rh = int(result_rh)
+            else:
                 result_rh = 0
-            self.log.info("Rebasehelper finish properly")
-
-        except Exception as ex:
-            self.log.info('builsys.py: Rebase helper failed with unknown reason. %s' % str(ex))
-        rh_stuff = rh_app.get_rebasehelper_data()
-        self.log.info(rh_stuff)
-        return result_rh, rh_stuff
-
-    def rebase_helper_checkers(self, new_version, old_task_id, new_task_id, tmp_dir):
-        argument = ['--buildtool', 'koji',
-                    '--non-interactive',
-                    '--builds-nowait',
-                    '--results-dir', tmp_dir,
-                    '--build-tasks',
-                    old_task_id + ',' + new_task_id,
-                    new_version,
-                    ]
-        rh_app = None
-        rh_stuff = None
-        try:
-            LoggerHelper.add_stream_handler(logger, logging.INFO)
-            cli = CLI(argument)
-            rh_app = Application(cli, *Application.setup(cli))
-            ret_code = rh_app.run()
-            if int(ret_code) != 0:
-                self.log.warn('Comparing package were not successful')
+            self.log.info("rebase-helper finished with exit code %d" % result_rh)
             rh_stuff = rh_app.get_rebasehelper_data()
-        except Exception:
-            self.log.exception('Compare packages failed.')
-        self.log.info(rh_stuff)
-
-        return rh_stuff
+            self.log.debug(rh_stuff)
+            return result_rh, rh_stuff
+        finally:
```

Comment:
This try/finally block is much larger than it needs to be. Can we move this finally up to right after``rh_app.run()``